Code/Source/RobotControl/RobotControl.h

```diff
@@ -26,15 +31,22 @@ namespace ROS2
     class RobotControl : public IRobotControl
     {
     public:
-        // TODO - pass component args (qos, topic name, etc) as editor-enabled serializable struct
-        void Activate(const AZ::Entity* entity, const AZStd::string& topicName, const QoS &qos) final
+        void Activate(const AZ::Entity* entity,
+                      const ControlConfiguration& controlConfiguration) final
         {
-            SetTargetComponent(entity);
+            m_controlConfiguration = controlConfiguration;
             m_active = true;
             if (!m_controlSubscription)
             {
+                auto ros2Frame = entity->FindComponent<ROS2FrameComponent>();
```

Comment:
It seems that entity argument is only used to get the Frame, which is only used to get namespacedTopic. Perhaps namespace could be passed as an argument instead? Or even inside the control configuration (as a non-serializable field)? Consider.

Reply:
I think passing the `AZ::Entity` is more semantically correct for the `Activate` method than just a string namespace. I would rather keep this generalization just to keep the code a little bit more organized.

This also keeps the RobotControl more flexible in case some other component dependencies came up in the future.